keras_nlp/tokenizers/word_piece_tokenizer_trainer_test.py

```diff
@@ -0,0 +1,326 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for Word Piece Tokenizer Trainer."""
+
+import os
+
+import tensorflow as tf
+
+from keras_nlp.tokenizers.word_piece_tokenizer_trainer import (
+    compute_word_piece_vocabulary,
+)
+
+
+class WordPieceTokenizerTrainerTest(tf.test.TestCase):
+    def test_dataset_input(self):
+        test_text = ["bat mat cat sat pat."]
+        test_output = [
+            "[PAD]",
+            "[CLS]",
+            "[SEP]",
+            "[UNK]",
+            "[MASK]",
+            ".",
+            "a",
+            "b",
+            "c",
+            "m",
+            "p",
+            "s",
+            "t",
+            "##at",
+            "##.",
+            "##a",
+            "##b",
+            "##c",
+            "##m",
+            "##p",
+            "##s",
+            "##t",
+        ]
+        data = tf.data.Dataset.from_tensor_slices(test_text)
+        vocab = compute_word_piece_vocabulary(
+            data,
+            10,
```

Comment:
This is mainly because the learner adds a "fixed" number of values such as each character + the required tokens, and also because the default minimum count for words is 10 (the lower bound) in the binary search, causing it to immediately exit. TLDR, this is a problem that only happens because the input data is too small, and shouldn't affect practical usage with larger vocab size and datasets. There are three potential solutions  I see for this:
1. Add a cutoff at the end by just slicing the vocab list, making sure that the vocab_size is at least as large as the required_tokens.
2. Manipulate the lower and upper bounds defaults for the learner to fit smaller sized datasets and potentially having to add some logic to "pick" the tokens that would be selected (since by default it just adds all the characters).
3. Make sure people don't put a vocab size that is too low when using this layer.
I would recommend either 1 or 3, since this layer is not useful for small inputs anyways, and there is no need for overengineering to support small inputs. Wdyt?

Reply:
I think 1 sounds good to me! Let's just slice down to vocabulary_size, and error if vocabulary_size < len(reserved tokens).